src/odb/src/db/dbITerm.cpp

```diff
@@ -666,4 +671,51 @@ void dbITerm::staSetVertexId(uint32_t id)
   iterm->_sta_vertex_id = id;
 }
 
+void dbITerm::setAccessPoint(dbMPin* pin, dbAccessPoint* ap)
+{
+  _dbITerm* iterm = (_dbITerm*) this;
+  if (ap != nullptr) {
+    iterm->aps_[pin->getImpl()->getOID()] = ap->getImpl()->getOID();
+    _dbAccessPoint* _ap = (_dbAccessPoint*) ap;
+    _ap->iterms_.push_back(iterm->getOID());
+  }
+  else
+    iterm->aps_[pin->getImpl()->getOID()] = dbId<_dbAccessPoint>();
+}
+
+std::vector<std::vector<dbAccessPoint*>> dbITerm::getAccessPoints() const
+{
+  _dbBlock* block = (_dbBlock*) getBlock();
+  auto mterm = getMTerm();
+  uint pin_access_idx = getInst()->getPinAccessIdx();
+  auto mpins = mterm->getMPins();
+  std::vector<std::vector<dbAccessPoint*>> aps(mpins.size());
+  uint i = 0;
+  for(auto mpin : mpins)
+  {
+    _dbMPin* pin = (_dbMPin*) mpin;
+    if(pin->aps_.size() > pin_access_idx)
+      for(auto id : pin->aps_[pin_access_idx])
+      {
+        aps[i].push_back((dbAccessPoint*) block->ap_tbl_->getPtr(id));
+      }
+    i++;
+  }
+  return aps;
+}
+
+std::vector<dbAccessPoint*> dbITerm::getPrefAccessPoints() const
```

Comment:
Similarly pin->ap map

Reply:
Wouldn't this be a bit repetitive?! you can get the dbMPin of the access point directly through getMPin(). And it is a 1-1 relation(each pin has 1 preferred access point at most).

Reply:
That makes sense so long as nobody is using the vector ordering to imply anything.